stl/inc/xlocnum

```diff
@@ -251,6 +251,230 @@ __PURE_APPDOMAIN_GLOBAL locale::id numpunct<_Elem>::id;
 #endif // __clang__
 #endif // !defined(_CRTBLD) || defined(CRTDLL2) || !defined(_DLL) || defined(_M_CEE_PURE)
 
+// TRANSITION, ABI: Sentinel value used by num_get::do_get() to enable correct "V2" behavior
+#define _ENABLE_V2_BEHAVIOR 1000000000
+
+template <class _Elem, class _InIt>
+int _Parse_fp_with_ios_base(char* const _Ac, _InIt& _First, _InIt& _Last, ios_base& _Iosbase, int* _Phexexp) {
+    // get floating-point field from [_First, _Last) into _Ac
+    const auto& _Punct_fac = _STD use_facet<numpunct<_Elem>>(_Iosbase.getloc());
+    char* _Ptr             = _Ac;
+
+    constexpr size_t _Offset_dec_digit_end = 10;
+    constexpr size_t _Offset_hex_digit_end = 22;
+    constexpr size_t _Offset_neg_sign      = 22;
+    constexpr size_t _Offset_pos_sign      = _Offset_neg_sign + 1;
+    constexpr size_t _Offset_upper_x       = 24;
+    constexpr size_t _Offset_lower_x       = _Offset_upper_x + 1;
+    constexpr size_t _Offset_upper_p       = 26;
+    constexpr size_t _Offset_lower_p       = _Offset_upper_p + 1;
+    constexpr size_t _Offset_upper_e       = 0xE;
+    constexpr size_t _Offset_lower_e       = 0xE + 0x10 - 10;
+    static constexpr char _Src[]           = "0123456789ABCDEFabcdef-+XxPp";
+    _Elem _Atoms[sizeof(_Src)];
+    const ctype<_Elem>& _Ctype_fac = _STD use_facet<ctype<_Elem>>(_Iosbase.getloc());
+    _Ctype_fac.widen(_STD begin(_Src), _STD end(_Src), _Atoms);
+
+    const _Elem _Positive_sign = _Atoms[_Offset_pos_sign];
+    const _Elem _Negative_sign = _Atoms[_Offset_neg_sign];
+    const _Elem _Zero_wc       = _Atoms[0];
+
+    if (_First != _Last) {
+        if (*_First == _Positive_sign) { // gather plus sign
+            *_Ptr++ = '+';
+            ++_First;
+        } else if (*_First == _Negative_sign) { // gather minus sign
+            *_Ptr++ = '-';
+            ++_First;
+        }
+    }
+
+    *_Ptr++ = '0'; // backstop carries from sticky bit
+
+    bool _Parse_hex                = false;
+    char _Initial_dec_leading_zero = '\0';

```

Comment:
Given that this is a count of leading zeroes, could be called `_Initial_dec_leading_zero_count` and initialized to `0`, not `'\0'` (since it's a number).

Also, what happens if we're parsing a hexadecimal number? Are we just okay with `000000000xBAFA`?

Reply:
For `"000000000xBAFA"`, currently the leading zeroes are accumulated, while `'x'` remains.

[[facet.num.get.virtuals]/3](https://eel.is/c++draft/facet.num.get.virtuals#3) says:
> If it is not discarded, then a check is made to determine if `c` is allowed as the next character of an input field of the conversion specifier returned by Stage 1. If so, it is accumulated.

I think current appoach is right since we already know that the `'x'` following `"000000000"` definitely results in an invalid input for `strtod`, given only leading `"0x"`/`"0X"` are allowed for hexadecimal representations.